### PR TITLE
Split

### PR DESCRIPTION
**Pull request by oluwandabira**
### Resolves

Issue #82 

### What's new

- Added a `pages` field to Window for all the pages.
- Changed Window.current to a string representing the current page map key.
- Added string key constants for each page.
- Switched the Wallets page to a self contained page that declares
 and handles it's own widgets (this page is still in its preliminary
 stage).
 - Other pages were left as is but will be switched over in pull requests.
 - Dialogs are controlled by the pages themselves.
* Error handling for slow wallet operations is not supported yet

### CODE REVIEW

File: ui/wallet_page.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+			},
+		}
+
+		page.dialog.LayoutIfActive(gtx, func() {
+			common.LayoutWithWallets(gtx, func() {
+				page.container.Layout(common.gtx, len(wdgs), func(i int) {
+					wdgs[i]()
+				})
+			})
+		})
```
Comment (oshorefueled) [inline ui/wallet_page.go]: The page layout should be wrapping the dialog layout. Although currently it still produces the intended result, the page layout in the dialog is counter-intuitive.

File: ui/wallet_page.go [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+				log.Error(err)
+			} else {
+				common.info.Wallets[*common.selectedWallet].Name = name
+				page.renaming = false
+			}
+		}
+
+		if page.delete.Clicked(gtx) {
+			page.dialog.SetDialog(page.dialog.deleteWallet(common))
+		}
```
Comment (oshorefueled) [inline ui/wallet_page.go]: The event handling conditions should be in a separate method, the result is a much cleaner code. To make sure every page has an event handling method, you can define an interface which every valid page must implement.

**Review by oshorefueled — CHANGES_REQUESTED** [review]

File: ui/wallet_page.go (hunk 2 above)
Comment (oluwandabira) [reply ui/wallet_page.go]: I actually prefer this to a separate method given that the only requirement a page has to satisfy now is returning a `layout.Widget` so adding an interface seems like unnecessary complexity.

File: ui/wallet_page.go (hunk 1 above)
Comment (oluwandabira) [reply ui/wallet_page.go]: It doesn't seem counter-intuitive to me since the dialog is not part of the page but rather a layer above it.

**Review by oshorefueled — APPROVED** [review]

**Pull request merged**